### PR TITLE
Adding more tests for geospatial capabilities

### PR DESCRIPTION
**Pull request by metonymic-smokey**
Signed-off-by: Aditi Ahuja <aditi.ahuja@couchbase.com>

### CODE REVIEW

File: search/searcher/search_intersects_test.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+		if err != nil {
+			t.Error(err)
+		}
+
+		indexReader, err := i.Reader()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer func() {
```
Comment (sreekanth-cb) [inline search/searcher/search_intersects_test.go]: why u need this defer func() here? You may close it after the delete.
Else it is a leak during the run of the tests

**Review by sreekanth-cb — COMMENTED** [review]
.

File: search/searcher/search_intersects_test.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+		Expected         []string
+	}{
+		{
+			QueryShape:       [][]float64{{3.0, 2.0}, {4.0, 2.0}},
+			DocShapeVertices: [][]float64{{3.0, 2.0}, {4.0, 2.0}},
+			DocShapeName:     "linestring1",
+			Desc:             "coincident linestrings",
+			Expected:         []string{"linestring1"},
+		},
+		{ // check this one
```
Comment (metonymic-smokey) [inline search/searcher/search_intersects_test.go]: @sreekanth-cb could you check this one out?

File: search/searcher/search_intersects_test.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+			Expected:         []string{"linestring1"},
+		},
+		{
+			QueryShape:       [][]float64{{1.0, 1.0}, {2.0, 2.0}},
+			DocShapeVertices: [][]float64{{1.5499860, 1.5501575}, {1.5, 1.5001714}},
+			DocShapeName:     "linestring1",
+			Desc:             "subline inside linestring",
+			Expected:         []string{"linestring1"},
+		},
+		{ // not as expected
```
Comment (metonymic-smokey) [inline search/searcher/search_intersects_test.go]: @sreekanth-cb and pls check this one out as well?
Thanks!

File: search/searcher/search_intersects_test.go [hunk 4]
```diff
@@ -92,6 +92,154 @@ func TestPointIntersects(t *testing.T) {
...
+
+func TestPointPolygonIntersects(t *testing.T) {
+	tests := []struct {
+		QueryShape       []float64
+		DocShapeVertices [][][]float64
+		DocShapeName     string
+		Desc             string
+		Expected         []string
+	}{
+		{ // check this one
```
Comment (metonymic-smokey) [inline search/searcher/search_intersects_test.go]: @sreekanth-cb these three cases were the ones not matching. Please let me know if this divergence is expected.
Thanks!

**Pull request closed without merging**